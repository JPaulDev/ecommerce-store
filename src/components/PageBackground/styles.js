import styled from 'styled-components';

export const Wrapper = styled.div`
  height: 100%;
  position: absolute;
  width: 100%;
  z-index: -100;
`;

export const Background = styled.div`
  background-color: var(--color-grey-9);
  background-image: url(${({ background }) => background});
  background-position: center top;
  background-repeat: no-repeat;
  height: 100vh;
  position: sticky;
  top: 0;
`;
