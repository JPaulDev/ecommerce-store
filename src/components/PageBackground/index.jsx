import * as Styled from './styles';

export default function PageBackground({ background }) {
  return (
    <Styled.Wrapper>
      <Styled.Background background={background} />
    </Styled.Wrapper>
  );
}
