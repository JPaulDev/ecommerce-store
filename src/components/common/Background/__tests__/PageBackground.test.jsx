import renderer from 'react-test-renderer';
import PageBackground from '../index';

describe('PageBackground component', () => {
  it('renders a container and a background image', () => {
    const tree = renderer.create(<PageBackground background />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
