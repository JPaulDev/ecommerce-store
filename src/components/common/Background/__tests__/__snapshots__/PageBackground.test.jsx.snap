// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBackground component renders a container and a background image 1`] = `
.c0 {
  height: 100%;
  position: absolute;
  width: 100%;
  z-index: -100;
}

.c1 {
  background-color: var(--color-grey-12);
  background-image: linear-gradient(var(--color-grey-15),var(--color-grey-15));
  background-position: center;
  background-repeat: no-repeat;
  background-size: 1294px 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
}

.c1 img {
  display: none;
}

@media screen and (min-width:1310px) {
  .c1 img {
    display: initial;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=85"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=85 1x, /_next/image?url=%2Fimg.jpg&w=48&q=85 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
  </div>
</div>
`;
